File: app/src/main/java/com/nrs/nsnik/notes/viewmodel/NoteViewModel.java
/*
 * Copyright (C) 2017 nsnikhil
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
 */

package com.nrs.nsnik.notes.viewmodel;

import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;
import android.arch.lifecycle.LiveData;

import com.nrs.nsnik.notes.MyApplication;
import com.nrs.nsnik.notes.data.NoteEntity;
import com.nrs.nsnik.notes.util.DbUtil;

import java.util.List;


public class NoteViewModel extends AndroidViewModel {

    private DbUtil mDbUtil;
    private LiveData<List<NoteEntity>> mNoteList;

    public NoteViewModel(Application application) {
        super(application);
        mDbUtil = ((MyApplication) application).getDbUtil();
        mNoteList = mDbUtil.getNoteList();
    }

    public LiveData<List<NoteEntity>> getNoteList() {
        return this.mNoteList;
    }

    public void insertNote(NoteEntity... noteEntities) {
        mDbUtil.insertNote(noteEntities);
    }

    public void updateNote(NoteEntity... noteEntities) {
        mDbUtil.updateNote(noteEntities);
    }

    public void deleteNote(NoteEntity... noteEntities) {
        mDbUtil.deleteNote(noteEntities);
    }

    public void deleteNoteByFolderName(String folderName) {
        mDbUtil.deleteNoteByFolderName(folderName);
    }

    public LiveData<NoteEntity> getNoteById(int id) {
        return mDbUtil.getNoteById(id);
    }

    public LiveData<List<NoteEntity>> getNoteByFolderName(String folderName) {
        return mDbUtil.getNoteByFolderName(folderName);
    }

    public LiveData<List<NoteEntity>> searchNote(String query) {
        return mDbUtil.searchNote(query);
    }

    public LiveData<List<NoteEntity>> getNoteByPin(int isPinned) {
        return mDbUtil.getNotesByPin(isPinned);
    }

    public LiveData<List<NoteEntity>> getNoteByLock(int isLocked) {
        return mDbUtil.getNotesByLock(isLocked);
    }

    public LiveData<List<NoteEntity>> getNoteByColor(String color) {
        return mDbUtil.getNotesByColor(color);
    }
}

File: app/src/main/java/com/nrs/nsnik/notes/viewmodel/NoteViewModel.kt
/*
 *     Notes  Copyright (C) 2018  Nikhil Soni
 *     This program comes with ABSOLUTELY NO WARRANTY; for details type `show w'.
 *     This is free software, and you are welcome to redistribute it
 *     under certain conditions; type `show c' for details.
 *
 * The hypothetical commands `show w' and `show c' should show the appropriate
 * parts of the General Public License.  Of course, your program's commands
 * might be different; for a GUI interface, you would use an "about box".
 *
 *   You should also get your employer (if you work as a programmer) or school,
 * if any, to sign a "copyright disclaimer" for the program, if necessary.
 * For more information on this, and how to apply and follow the GNU GPL, see
 * <http://www.gnu.org/licenses/>.
 *
 *   The GNU General Public License does not permit incorporating your program
 * into proprietary programs.  If your program is a subroutine library, you
 * may consider it more useful to permit linking proprietary applications with
 * the library.  If this is what you want to do, use the GNU Lesser General
 * Public License instead of this License.  But first, please read
 * <http://www.gnu.org/philosophy/why-not-lgpl.html>.
 */

package com.nrs.nsnik.notes.viewmodel

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData

import com.nrs.nsnik.notes.MyApplication
import com.nrs.nsnik.notes.data.NoteEntity
import com.nrs.nsnik.notes.util.DbUtil


class NoteViewModel(application: Application) : AndroidViewModel(application) {

    private val mDbUtil: DbUtil = (application as MyApplication).dbUtil

    fun insertNote(vararg noteEntities: NoteEntity) = mDbUtil.insertNote(*noteEntities)

    fun updateNote(vararg noteEntities: NoteEntity) = mDbUtil.updateNote(*noteEntities)

    fun deleteNote(vararg noteEntities: NoteEntity) = mDbUtil.deleteNote(*noteEntities)

    fun deleteNoteByFolderName(folderName: String) = mDbUtil.deleteNoteByFolderName(folderName)

    fun getNoteById(id: Int): LiveData<NoteEntity> = mDbUtil.getNoteById(id)

    fun getNoteByFolderName(folderName: String): LiveData<List<NoteEntity>> = mDbUtil.getNoteByFolderName(folderName)

    fun getNoteByFolderNameNoPinNoLock(folderName: String): LiveData<List<NoteEntity>> = mDbUtil.getNoteByFolderNameNoPinNoLock(folderName)

    fun getNoteByFolderNamePinNoLock(folderName: String): LiveData<List<NoteEntity>> = mDbUtil.getNoteByFolderNamePinNoLock(folderName)

    fun getNoteByFolderNameNoPinLock(folderName: String): LiveData<List<NoteEntity>> = mDbUtil.getNoteByFolderNameNoPinLock(folderName)

    fun getNoteByFolderNamePinLock(folderName: String): LiveData<List<NoteEntity>> = mDbUtil.getNoteByFolderNamePinLock(folderName)

    fun getNoteByFolderNameOrdered(folderName: String): LiveData<List<NoteEntity>> = mDbUtil.getNoteByFolderNameOrdered(folderName)

    fun getNoteByFolderNameOrderedLock(folderName: String): LiveData<List<NoteEntity>> = mDbUtil.getNoteByFolderNameOrderedLock(folderName)

    fun searchNote(query: String): LiveData<List<NoteEntity>> = mDbUtil.searchNote(query)

    fun getNoteByPin(isPinned: Int): LiveData<List<NoteEntity>> = mDbUtil.getNotesByPin(isPinned)

    fun getNoteByLock(isLocked: Int): LiveData<List<NoteEntity>> = mDbUtil.getNotesByLock(isLocked)

    fun getNoteByColor(color: String): LiveData<List<NoteEntity>> = mDbUtil.getNotesByColor(color)

    fun changeNotePinStatus(id: Int, pin: Int) = mDbUtil.changeNotePinStatus(id, pin)

    fun changeNoteLockStatus(id: Int, lock: Int) = mDbUtil.changeNoteLockStatus(id, lock)

    fun changeNoteFolder(id: Int, folderName: String) = mDbUtil.changeNoteFolder(id, folderName)
}